Doc/reference/datamodel.rst

```diff
@@ -2043,10 +2044,10 @@ Basic customization
 
 .. method:: object.__str__(self)
 
-   Called by :func:`str(object) <str>` and the built-in functions
-   :func:`format` and :func:`print` to compute the "informal" or nicely
+   Called by :func:`str(object) <str>`, the default :meth:`__format__` implementation,
+   and the built-in function :func:`print`, to compute the "informal" or nicely
    printable string representation of an object.  The return value must be a
-   :ref:`string <textseq>` object.
+   :class:`str` object.
```

Comment:
The link went nicely to the `str` section, a few paragraph above the target of this changed link (the str function definition).

If the goal of the change is to say «str» and not «string», to avoid imprecision or suggesting that there is more than one string type, I see the value! Then let’s do this:

```suggestion
   :ref:`str <textseq>` object.
```